core-web/libs/portlets/dot-ema/src/lib/components/edit-ema-persona-selector/edit-ema-persona-selector.component.spec.ts

```diff
@@ -0,0 +1,112 @@
+                            entity: [DEFAULT_PERSONA],
+                            pagination: {
+                                totalEntries: 1,
+                                page: 1,
+                                perPage: 10
+                            }
+                        });
+                    }
+                }
+            }
+        ]
+    });
+
+    beforeEach(() => {
+        spectator = createComponent({
+            props: {
+                selectedPersona: {
+                    ...DEFAULT_PERSONA,
+                    identifier: 'some test'
+                },
+                pageID: '123'
+            }
+        });
+
+        component = spectator.component;
+    });
+
+    describe('DOM', () => {
+        it('should have an autocomplete', () => {
+            expect(spectator.query(byTestId('persona-selector-autocomplete'))).not.toBeNull();
+        });
+    });
+
+    describe('events', () => {
+        it('should trigger onSelect when a persona is selected', () => {
+            const onSelect = jest.spyOn(component, 'onSelect');
+            spectator.triggerEventHandler('p-autocomplete', 'onSelect', {});
+            expect(onSelect).toHaveBeenCalled();
+        });
+
+        it('should emit the default persona when the selected emits the default persona', () => {
+            const selectedSpy = jest.spyOn(component.selected, 'emit');
+            spectator.triggerEventHandler('p-autocomplete', 'onSelect', DEFAULT_PERSONA);
+            expect(selectedSpy).toHaveBeenCalledWith(DEFAULT_PERSONA);
+        });
+
+        it('should not emit the selected persona when it is already selected', () => {
+            const selectedSpy = jest.spyOn(component.selected, 'emit');
+            spectator.triggerEventHandler('p-autocomplete', 'onSelect', {
+                identifier: 'some test'
+            });
+            expect(selectedSpy).not.toHaveBeenCalled();
+        });
+
+        it("should open a confimartion dialog when the selected persona isn't personalized", () => {
+            const confirmationService = spectator.inject(ConfirmationService);
+
+            const confirmationMock = jest.spyOn(confirmationService, 'confirm');
+
+            spectator.triggerEventHandler('p-autocomplete', 'onSelect', {
+                identifier: 'some test 2',
+                personalized: false
+            });
+
+            expect(confirmationMock).toHaveBeenCalled();
+        });
+
+        it('should emit the selected persona when it is personalized', () => {
+            const selectedSpy = jest.spyOn(component.selected, 'emit');
+
+            spectator.triggerEventHandler('p-autocomplete', 'onSelect', {
+                identifier: 'some test 2',
+                personalized: true
+            });
+
+            expect(selectedSpy).toHaveBeenCalledWith({
+                identifier: 'some test 2',
+                personalized: true
+            });
+        });
+    });
```

Comment:
Is not the same test as in line 77?

Reply:
Not really, If it's not the default it checks if its personalized to open or not the confirm dialog.